src/library/scala/collection/IterableOnce.scala

```diff
@@ -650,7 +666,7 @@ trait IterableOnceOps[+A, +CC[_], +C] extends Any { this: IterableOnce[A] =>
    *            are the elements of this $coll.
     *           Returns `z` if this $coll is empty.
     */
-  def foldLeft[B](z: B)(op: (B, A) => B): B = {
+  def foldLeft[B](z: B)(op: (B, A) => B): B = if (this.isInstanceOf[IndexedSeq[_]]) foldl(this.asInstanceOf[IndexedSeq[A]], 0, z, op) else {
```

Comment:
`this match` to make it like the others?

Reply:
ichoran made me feel foolish for using match all the time.